### PR TITLE
[Bombastic Perks] Add some new perks

### PR DESCRIPTION
**Pull request by Standing-Storm**
#### Summary
Mods "[Bombastic Perks] Add some new perks"

#### Purpose of change

Had an idea for a few new perks.

#### Describe the solution

Add three perks:

1. Vengeful: Inspired by the Isolation Protocol ability but much less powerful.  For two turns after you're hit, you do 15% more damage.
2. Quick Recovery: A 75% chance to stand up immediately if you're downed.
3. Easy Sleeper: Inspired by my sister. Adds +15 to SLEEPY, making it much easier to fall asleep under any circumstances.

#### Testing

Everything is selectable and works.

### CODE REVIEW

**Comment by bombasticSlacks** [comment]
Hey sorry I didn't get to review this before merging, I'm going to make a few notes as comments, no pressure to fix them I'll compile and do it sometime soon if you don't 😄 

Thanks for contributing perks!

File: data/mods/BombasticPerks/perkdata/quick_recovery.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_PERK_QUICK_RECOVERY_INIT",
+    "eoc_type": "EVENT",
+    "required_event": "character_gains_effect",
+    "condition": {
+      "and": [
+        { "u_has_trait": "perk_quick_recovery" },
+        { "x_in_y_chance": { "x": 3, "y": 4 } },
+        { "or": [ { "compare_string": [ "downed", { "context_val": "effect" } ] } ] }
```
Comment (bombasticSlacks) [inline data/mods/BombasticPerks/perkdata/quick_recovery.json:11]: I think this "or" isn't necessary.

File: data/mods/BombasticPerks/perkdata/vengeful.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
+[
+  {
+    "type": "effect_type",
+    "id": "effect_perk_vengeful",
+    "rating": "good",
+    "int_dur_factor": "1 s",
+    "max_duration": "2 s",
+    "show_in_info": false
```
Comment (bombasticSlacks) [inline data/mods/BombasticPerks/perkdata/vengeful.json:8]: Probably worth showing when active

File: data/mods/BombasticPerks/perkmenu.json [hunk 3]
```diff
@@ -355,6 +355,24 @@
...
+        "text": "Gain [<trait_name:perk_vengeful>]",
+        "effect": [
+          { "set_string_var": "<trait_name:perk_vengeful>", "target_var": { "context_val": "trait_name" } },
+          {
+            "set_string_var": "<trait_description:perk_vengeful>",
+            "target_var": { "context_val": "trait_description" }
+          },
+          { "set_string_var": "perk_vengeful", "target_var": { "context_val": "trait_id" } },
+          {
+            "set_string_var": "Requires handguns 2",
```
Comment (bombasticSlacks) [inline data/mods/BombasticPerks/perkmenu.json:369]: This is a typo I think

Comment (Standing-Storm) [reply data/mods/BombasticPerks/perkmenu.json:369]: You're right--I removed it from the actual requirements but not from the listed requirements

File: data/mods/BombasticPerks/perkdata/quick_recovery.json (hunk 1 above)
Comment (Standing-Storm) [reply data/mods/BombasticPerks/perkdata/quick_recovery.json:11]: Originally it also did "or dazed" but I decided that was better than "kinda okay" so I removed it but not the or. I'll take that out.

**Pull request merged**